Frameworks/Foundation/NSException.mm

```diff
@@ -216,7 +244,18 @@ - (HRESULT)_hresult {
         return [hresultValue unsignedIntValue];
     }
 
-    TraceWarning(L"NSException", L"Called -hresult on an NSException without an HRESULT!");
-    return E_UNEXPECTED;
+    return [[self class] _HRESULTForExceptionName:self.name];
+}
+
+- (void)_processException {
+    HRESULT hr = [self _hresult];
+    HSTRING_HEADER header;
+    HSTRING hsReason = nullptr;
+    
+    if(self.reason != nil) {
+        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
+        WindowsCreateStringReference(reason, wcslen(reason), &header, &hsReason);
```

Comment:
And it is less code....
Again, feeling stupid I missed the 

"Try ***using**** [https://msdn.microsoft.com/en-us/library/hh771147.aspx](https://msdn.microsoft.com/en-us/library/hh771147.aspx) as mentioned in previous iteration."


---
In reply to: [93377905](https://github.com/Microsoft/WinObjC/pull/1570#discussion_r93377905) [](ancestors = 93377905,93301103,93299684,93297284,93125423)